ui/app/components/create-policy-modal.js

```diff
@@ -0,0 +1,93 @@
+import Component from '@glimmer/component';
+import { action } from '@ember/object';
+import { inject as service } from '@ember/service';
+import { tracked } from '@glimmer/tracking';
+
+/**
+ * @module CreatePolicyModal
+ * CreatePolicyModal components are meant to render within a modal for creating a new policy of unknown type.
+ *
+ * @example
+ *  <CreatePolicyModal
+ *    @nameInput="new-item-name"
+ *    @onSave={{this.closeModal}}
+ *    @onCancel={{this.closeModal}}
+ *  />
+ * ```
+ * @callback onCancel - callback triggered when cancel button is clicked
+ * @callback onSave - callback triggered when save button is clicked
+ * @param {string} nameInput - the name of the newly created policy
+ */
+
+export default class PolicyFormComponent extends Component {
+  @service store;
+  @service version;
+  @service flashMessages;
+
+  @tracked policy = null;
+  @tracked showExamplePolicy = false;
+
+  get policyOptions() {
+    return [
+      { label: 'ACL Policy', value: 'acl', isDisabled: false },
+      { label: 'Role Governing Policy', value: 'rgp', isDisabled: !this.version.hasSentinel },
+    ];
+  }
+  // formatting here is purposeful so that whitespace renders correctly in JsonEditor
+  policyTemplates = {
+    acl: `
+# Grant 'create', 'read' , 'update', and ‘list’ permission
+# to paths prefixed by 'secret/*'
+path "secret/*" {
+  capabilities = [ "create", "read", "update", "list" ]
+}
+
+# Even though we allowed secret/*, this line explicitly denies
+# secret/super-secret. This takes precedence.
+path "secret/super-secret" {
+  capabilities = ["deny"]
+}
+`,
+    rgp: `
+# Import strings library that exposes common string operations
+import "strings"
+
+# Conditional rule (precond) checks the incoming request endpoint
+# targeted to sys/policies/acl/admin
+precond = rule {
+    strings.has_prefix(request.path, "sys/policies/admin")
+}
+
+# Vault checks to see if the request was made by an entity
+# named James Thomas or Team Lead role defined as its metadata
+main = rule when precond {
+    identity.entity.metadata.role is "Team Lead" or
+      identity.entity.name is "James Thomas"
+}
+`,
+  };
+
+  @action
+  setPolicyType(type) {
+    // Create form model once type is chosen
+    // cleanup any model argument before firing parent action to create a new record in identity/edit-form.js
+    if (this.policy) this.cleanup();
+    const model = this.store.createRecord(`policy/${type}`, { name: this.args.nameInput });
+    this.policy = model;
+  }
+
+  @action onSave(policyModel) {
+    const { name, policyType, isNew } = policyModel;
```

Comment:
Great catch! Do we think the flash message trigger should live within the form itself, since it can trigger it based on whether the model was new when save was clicked? I think that keeps the `onSave` handler clean. Or we can add another param to the `onSave` handler.